backend/src/main/java/com/votogether/domain/post/service/PostCommandService.java

```diff
@@ -0,0 +1,360 @@
+    private void updatePostOptions(final Post post, final List<PostOptionUpdateRequest> postOptionUpdates) {
+        final List<PostOption> postOptions = new ArrayList<>(post.getPostOptions());
+        final Set<Long> postOptionIds = postOptions.stream()
+                .map(PostOption::getId)
+                .collect(Collectors.toSet());
+        final List<PostOption> removePostOptions = extractPostOptionRemoves(postOptions, postOptionUpdates);
+        final Map<Long, PostOptionUpdateRequest> updateOptionGroup =
+                extractPostOptionUpdates(postOptionIds, postOptionUpdates);
+        final List<PostOptionUpdateRequest> addPostOptions =
+                extractPostOptionAdds(postOptionIds, postOptionUpdates);
+
+        removeOriginPostOptions(post, removePostOptions);
+        updateOriginPostOptions(post, updateOptionGroup);
+        addNewPostOptions(post, addPostOptions, post.getPostOptions().size());
+    }
+
+    private List<PostOption> extractPostOptionRemoves(
+            final List<PostOption> postOptions,
+            final List<PostOptionUpdateRequest> postOptionUpdates
+    ) {
+        final Set<Long> updateOptionIds = postOptionUpdates.stream()
+                .map(PostOptionUpdateRequest::getId)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+        return postOptions.stream()
+                .filter(postOption -> !updateOptionIds.contains(postOption.getId()))
+                .toList();
+    }
+
+    private Map<Long, PostOptionUpdateRequest> extractPostOptionUpdates(
+            final Set<Long> postOptionIds,
+            final List<PostOptionUpdateRequest> postOptionUpdates
+    ) {
+        return postOptionUpdates.stream()
+                .filter(postOptionUpdate ->
+                        Objects.nonNull(postOptionUpdate.getId()) && postOptionIds.contains(postOptionUpdate.getId()))
+                .collect(Collectors.toMap(
+                        PostOptionUpdateRequest::getId,
+                        postOptionUpdate -> postOptionUpdate,
+                        (exist, replace) -> {
+                            throw new BadRequestException(PostOptionExceptionType.DUPLICATE_UPDATE_POST_OPTION);
+                        },
+                        HashMap::new
+                ));
+    }
+
+    private List<PostOptionUpdateRequest> extractPostOptionAdds(
+            final Set<Long> postOptionIds,
+            final List<PostOptionUpdateRequest> postOptionUpdates
+    ) {
+        final boolean hasInvalidId = postOptionUpdates.stream()
+                .map(PostOptionUpdateRequest::getId)
+                .anyMatch(id -> id != null && !postOptionIds.contains(id));
+
+        if (hasInvalidId) {
+            throw new BadRequestException(PostOptionExceptionType.UNRELATED_POST_OPTION);
+        }
+
+        return postOptionUpdates.stream()
+                .filter(postOptionUpdate -> postOptionUpdate.getId() == null)
+                .toList();
+    }
```

Comment:
```P2```
```suggestion
private List<PostOptionUpdateRequest> extractPostOptionAdds(
        final Set<Long> postOptionIds,
        final List<PostOptionUpdateRequest> postOptionUpdates
) {
    validatePostOptionIds(postOptionIds, postOptionUpdates);

    return postOptionUpdates.stream()
            .filter(postOptionUpdate -> postOptionUpdate.getId() == null)
            .toList();
}

private void validatePostOptionIds(
        final Set<Long> postOptionIds,
        final List<PostOptionUpdateRequest> postOptionUpdates
) {
    boolean hasInvalidId = postOptionUpdates.stream()
            .map(PostOptionUpdateRequest::getId)
            .anyMatch(id -> id != null && !postOptionIds.contains(id));

    if (hasInvalidId) {
        throw new BadRequestException(PostOptionExceptionType.UNRELATED_POST_OPTION);
    }
}
```

이런 식으로 분리해준다면 메인 메서드는 주된 로직에만 더 집중하고 유효성 검증 로직도 따로 분리하여 더 명확해 질 수 있을 것 같은데 어떻게 생각하시나요 :)

Reply:
혹은 이러한 방식도 생각해볼 수 있을 것 같아요.
```suggestion
private List<PostOptionUpdateRequest> extractPostOptionAdds(
        final Set<Long> postOptionIds,
        final List<PostOptionUpdateRequest> postOptionUpdates
) {
    validatePostOptionIds(postOptionIds, postOptionUpdates);

    return postOptionUpdates.stream()
            .filter(postOptionUpdate -> postOptionUpdate.getId() == null)
            .toList();
}

private void validatePostOptionIds(
        final Set<Long> postOptionIds,
        final List<PostOptionUpdateRequest> postOptionUpdates
) {
    postOptionUpdates.stream()
            .map(PostOptionUpdateRequest::getId)
            .filter(Objects::nonNull)
            .filter(id -> !postOptionIds.contains(id))
            .findFirst()
            .ifPresent(invalidId -> {
                throw new BadRequestException(PostOptionExceptionType.UNRELATED_POST_OPTION);
            });
}
```

Reply:
스트림 활용을 너무 잘 하시네요 👍👍